### ISSUE TITLE
linux/arm64 编译报错

### ISSUE THREAD

**Issue by xuyang2** [emmansun/gmsm#11]
```
$ GOOS=linux GOARCH=arm64 go build ./...
# github.com/emmansun/gmsm/sm4
asm: illegal combination: 00308 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:256)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00428 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:264)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00548 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:272)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00668 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:280)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00308 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:256)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00428 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:264)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00548 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:272)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00668 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:280)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00308 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:256)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00428 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:264)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00548 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:272)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00668 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:280)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00308 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:256)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00428 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:264)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00548 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:272)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00668 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:280)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00244 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:342)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00364 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:350)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00484 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:358)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00604 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:366)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00244 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:342)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00364 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:350)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00484 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:358)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00604 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:366)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00244 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:342)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00364 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:350)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00484 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:358)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00604 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:366)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00244 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:342)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00364 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:350)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00484 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:358)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: illegal combination: 00604 (/code/github.com/emmansun/gmsm/sm4/asm_arm64.s:366)   VDUP    R19, V0.S4 REG NONE NONE ARNG, 7 7
asm: assembly failed
```

```
$ git rev-parse --verify HEAD
acabed56bd8c6e45ab58817cdf1af5cb145f8930
```

**Comment by emmansun** [comment]
关于ARM64，我只测试了[Go: 1.17.x + arm64-graviton2](https://app.travis-ci.com/github/emmansun/gmsm/jobs/554841941)

**Comment by emmansun** [comment]
[cmd/asm: complete the support for VDUP on arm64](https://github.com/golang/go/commit/9136d958ab258bc4f128c8582ab713c482ec33ed)

**Comment by emmansun** [comment]
released in v0.9.1
